### PR TITLE
Update Go to 1.22+

### PR DESCRIPTION
**Pull request by roman-khimov**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module github.com/nspcc-dev/rfc6979
 
-go 1.21
+go 1.22
```
Comment (carpawell) [inline go.mod:3]: why do you close the issue with _this_ commit?

File: .github/workflows/run_tests.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
 
   test_cover:
     name: Coverage
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
```
Comment (carpawell) [inline .github/workflows/run_tests.yml:27]: can be just latest?

**Review by carpawell — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (roman-khimov) [reply go.mod:3]: Because it updates to go 1.22.

File: .github/workflows/run_tests.yml (hunk 2 above)
Comment (roman-khimov) [reply .github/workflows/run_tests.yml:27]: It was a fixed version, it is a fixed version.

File: go.mod (hunk 1 above)
Comment (carpawell) [reply go.mod:3]: the issue has a reference to another issue which is a checklist with some requirements for updating

File: .github/workflows/run_tests.yml (hunk 2 above)
Comment (carpawell) [reply .github/workflows/run_tests.yml:27]: yes, so this commit might not even exist

**Pull request merged**